snuba/utils/streams/consumer.py

```diff
@@ -137,6 +120,7 @@ class KafkaConsumer:
     def __init__(
         self,
         configuration: Mapping[str, Any],
+        codec: Codec[Payload, TPayload],
```

Comment:
I like the idea that KafkaConsumer specifies the encoded payload type but lets the user have their own codec with their own decoded type, though the naming seems confusing `Payload` vs `TPayload`. What about renaming Payload into something Kafka specific so this is clear and easy to read? Anyway its attribute are already mostly Kafka specific.

Reply:
Makes sense to me, how about `KafkaPayload` or `KafkaMessagePayload`?

Reply:
Did this here 1711b6c6a455d7af12122926c5b87eae7554fa01.